File: pset6/readability/readability.py
def main():

    text = input("Text :")

    textparser(text)

def textparser(s):

    l = len(s)
    letters = 0
    words = 0
    sentences = 0

    for i in range(0, l, 1):

        if s[i] == ' ':
            words += 1

        elif s[i] in{'.','?','!'}:
            sentences += 1

        elif s[i].isupper or s[i].islower:
            letters += 1

    print(f"{letters} letters")
    print(f"{words} words")
    print(f"{sentences} sentences")

    letters = letters * 100/words
    sentences = sentences * 100/words

    textgrade = 0.0588 * letters - 0.296 * sentences - 15.8

    if textgrade < 1:
        print("Grade 1")
    elif textgrade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {int(textgrade)}")

main()

File: pset2/Readability/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>

void textparser(string text);

int main(void){

    //Get text from user
    string text = get_string("Text :");

    //Call parsing function
    textparser(text);

}

void textparser(string text){

    int l = strlen(text);
    int letters = 0;
    int words = 1;
    int sentences = 0;

    //Parse the word and count words letter and sentences depending on the character encountered
    for(int i = 0; i < l; i++){

        if(text[i] == ' '){
            words++;
        }else if(text[i] == '.'|| text[i] == '?' || text[i] == '!'){
            sentences++;
        }else if(isupper(text[i]) || islower(text[i])){
            letters++;
        }else{}
    }
    printf("\nLetters : %i \n", letters);
    printf("Words : %i \n", words);
    printf("Sentences : %i \n\n", sentences);

    //Get average number of letters and sentences per 100 words
    letters = letters * 100/words;
    sentences  = sentences * 100/words;

    //calculate the grade of the text with the Cordman-Liau formula
    float textgrade = 0.0588 * letters - 0.296 * sentences - 15.8;

    //Print the result
    if(textgrade < 1){
        printf("Grade 1 \n");
    }else if(textgrade > 16){
        printf("Grade 16+ \n");
    }else{
        printf("Grade %f \n", round(textgrade));
    }
}